core/src/test/scala/org/apache/spark/SparkContextSuite.scala

```diff
@@ -1285,6 +1285,30 @@ class SparkContextSuite extends SparkFunSuite with LocalSparkContext with Eventu
       }
     }
   }
+
+  test("SPARK-35691: addFile/addJar/addDirectory should put CanonicalFile") {
+    withTempDir { dir =>
+      try {
+        val sep = File.separator
+        val tmpDir = Utils.createTempDir(dir.getAbsolutePath + sep + "." + sep + ".")
+        val tmpJar = File.createTempFile("test", ".jar", tmpDir)
+        val tmpFile = File.createTempFile("test", ".txt", tmpDir)
+
+        sc = new SparkContext(new SparkConf().setAppName("test").setMaster("local"))
```

Comment:
`NettyStreamManager.addJar` is not performed if a test runs with `local` mode, meaning this doesn't test the change of `addFile`.

Reply:
Make scene. I added a new test to `RpcEnvSuite.scala`.

Reply:
I think you don't need to modify `RpcEnvSuite`.
How about `setMaster("local-cluster[...]")?

Reply:
Make scene, Thanks.

Reply:
Now we set `setMaster("local-cluster[...]")`, we don't need to add new assertions to `RpcEnvSuite.scala`.

Reply:
@sarutak Thanks for the review. Removed it.